### PR TITLE
fix: do not throw when there's no sidenav.

### LINKED ISSUE #269: MdSidenavlayout should not throw when no MdSidenav is present
* **Do you want to request a *feature* or report a *bug*?**

Bug

* **What is the current behavior?**

If I have an `MdSidenavLayout` without an `MdSidenav` child, an exception will be raised. For instance, I wanted to avoid pre-compiling sidenav content in a pre-generated html document, so I threw an ngIf on the md-sidenav to hide it, which caused an error.

* **Which version of Angular and Material, and which browser and OS does this issue affect?**

Definitely alpha.1, and I think this is still present in alpha.2

### PR DESCRIPTION
**Pull request by hansl**
Fixes #269.

### CODE REVIEW

**Comment by hansl** [comment]
@jelbourn @jeffbcross

File: src/components/sidenav/sidenav.ts [hunk 1]
```diff
@@ -286,9 +281,6 @@ export class MdSidenavLayout implements AfterContentInit {
    */
   private _validateDrawers() {
     this._start = this._end = null;
-    if (this._sidenavs.length === 0) {
-      throw new MdMissingSidenavException();
-    }
```
Comment (jelbourn) [inline src/components/sidenav/sidenav.ts]: Add a test? While the code isn't explicitly throwing an exception when a sidenav is missing, having a test would guard against accidentally throwing an error if the sidenav is missing (such as, for example, a null dereference).

Comment (hansl) [inline src/components/sidenav/sidenav.ts]: Donerino.

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**